test/parallel/test-single-executable-application.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+const common = require('../common');
+
+// This tests the creation of a single executable application.
+
+const tmpdir = require('../common/tmpdir');
+const { copyFileSync, readFileSync, writeFileSync } = require('fs');
+const { execSync } = require('child_process');
+const { join } = require('path');
+const { strictEqual } = require('assert');
+
+if (!process.config.variables.single_executable_application)
+  common.skip('Single Executable Application support has been disabled.');
+
+if (!['darwin', 'win32', 'linux'].includes(process.platform))
+  common.skip(`Unsupported platform ${process.platform}.`);
+
+if (process.config.variables.asan)
+  common.skip('Running the resultant binary fails with `Segmentation fault (core dumped)`.');
+
+if (process.platform === 'linux' && process.config.variables.is_debug === 1)
+  common.skip('Running the resultant binary fails with `Couldn\'t read target executable"`.');
+
+if (process.config.variables.node_shared)
+  common.skip('Running the resultant binary fails with ' +
+    '`/home/iojs/node-tmp/.tmp.2366/sea: error while loading shared libraries: ' +
+    'libnode.so.112: cannot open shared object file: No such file or directory`.');
+
+if (!process.config.variables.node_use_openssl || process.config.variables.node_shared_openssl)
+  common.skip('Running the resultant binary fails with `Node.js is not compiled with OpenSSL crypto support`.');
```

Comment:
Do you know why this matters? I’m surprised, given that there aren’t any interactions with OpenSSL afaict.

Reply:
I'm not sure currently but I think this originates from the resource injection part, i.e., LIEF because it essentially rewrites the entire binary, so it's possible that it misses some things here and there.

Reply:
… That sounds scary :slightly_smiling_face:

Reply:
I agree 😅. If this PR lands, it should be easier for us to debug these issues further without having to build Node.js from source for all the failing configurations.

Reply:
Do you think this issue should be fixed before landing?

Reply:
My 2 cents is to add is that I think the `scary` part is in terms of the tool used to inject versus what is is being added as part of this PR.  In other words is it more of a failure in thet `test` infra we currently have versus what is going to be in Node.js itself. I mention that as it makes me a bit more comfortable having this land in addition to it being Experimental. If we had different levels of experimental as was being discussed, this would likely be at the earliest stage.  I do agree that having it land will help in terms of improving/making it easier to debug.